From cbb380423b2038797dc438d886de7c25eeac78a6 Mon Sep 17 00:00:00 2001
From: Zhao Chenhui <chenhui.zhao@freescale.com>
Date: Fri, 23 Mar 2012 17:03:39 +0800
Subject: [PATCH 192/245] usb/fsl: enable USB wakeup

To enable USB wakeup feature,
 * Add a interrupt flag IRQF_NO_SUSPEND to avoid the USB interrupt
   disabled when suspending.
 * Set the USB module as a wakeup source.

Change-Id: I85b1b538175ded6e6f56f701d76f49da9884178f
Signed-off-by: Zhao Chenhui <chenhui.zhao@freescale.com>
Signed-off-by: Li Yang <leoli@freescale.com>
Reviewed-on: http://git.am.freescale.net:8181/1448
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
---
 drivers/usb/host/ehci-fsl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/host/ehci-fsl.c b/drivers/usb/host/ehci-fsl.c
index ab4eee3..e5d2516 100644
--- a/drivers/usb/host/ehci-fsl.c
+++ b/drivers/usb/host/ehci-fsl.c
@@ -121,7 +121,7 @@ static int usb_hcd_fsl_probe(const struct hc_driver *driver,
 
 	/* Don't need to set host mode here. It will be done by tdi_reset() */
 
-	retval = usb_add_hcd(hcd, irq, IRQF_SHARED);
+	retval = usb_add_hcd(hcd, irq, IRQF_SHARED | IRQF_NO_SUSPEND);
 	if (retval != 0)
 		goto err2;
 	device_wakeup_enable(hcd->self.controller);
-- 
2.1.0.27.g96db324

